Casks/charles.rb

```diff
@@ -3,11 +3,15 @@
   sha256 "81b779b352ce8ed588009472ae78f28c5f19613bec6d66f6f92703cdfb7df033"
 
   url "https://www.charlesproxy.com/assets/release/#{version}/charles-proxy-#{version}.dmg"
-  appcast "https://www.charlesproxy.com/latest.do"
   name "Charles"
   desc "Web debugging Proxy application"
   homepage "https://www.charlesproxy.com/"
 
+  livecheck do
+    url "https://www.charlesproxy.com/latest.do"
+    regex(/\d+(?:\.\d+)*/)
```

Comment:
```suggestion
    regex(/v?(\d+(?:\.\d+)+)/i)
```

We use a capture group around the version in regexes.